cmd/k8s-operator/proxygroup.go

```diff
@@ -230,15 +242,36 @@ func validateProxyClassForPG(logger *zap.SugaredLogger, pg *tsapi.ProxyGroup, pc
 	}
 }
 
-func (r *ProxyGroupReconciler) maybeProvision(ctx context.Context, pg *tsapi.ProxyGroup, proxyClass *tsapi.ProxyClass) error {
+func (r *ProxyGroupReconciler) maybeProvision(ctx context.Context, pg *tsapi.ProxyGroup, proxyClass *tsapi.ProxyClass) (done bool, err error) {
 	logger := r.logger(pg.Name)
 	r.mu.Lock()
 	r.ensureAddedToGaugeForProxyGroup(pg)
 	r.mu.Unlock()
 
-	if err := r.ensureConfigSecretsCreated(ctx, pg, proxyClass); err != nil {
-		return fmt.Errorf("error provisioning config Secrets: %w", err)
+	svcToNodePorts := make(map[string]uint16)
+	var tailscaledPort *uint16
+	if proxyClass != nil && proxyClass.Spec.StaticEndpoints != nil {
+		svcToNodePorts, tailscaledPort, err = r.ensureNodePortServiceCreated(ctx, pg, proxyClass)
+		if err != nil {
+			wrappedErr := fmt.Errorf("error provisioning NodePort Services for static endpoints: %w", err)
+			var allocatePortErr *allocatePortsErr
+			if errors.As(err, &allocatePortErr) {
+				return true, wrappedErr
```

Comment:
@ChaosInTheCRD I think thsi logic should be the other way round? If we fail to allocate ports (not enough etc) we want to return false and not an error

Reply:
This was resolved offline

Reply:
I've gone ahead and changed it to the way you have proposed @irbekrm, but there are still shortcomings to it and really this is something that we should be handling in subsequent PRs.

In practice, adding events into `maybeProvision` was looking like it was going to involve a lot of copying and pasting of code, so I opted for an extra method on the reconciler, and keeping the deep copy check of the two ProxyGroup status's out of `maybeProvision` entirely (didn't feel right feeding `oldPGStatus` in there.

Overall, the way we do this for each reconciler is very inconsistent so I think we should take this as a note to explore standardization of one solid approach across them all along the line 😄.